### DISCUSSION TITLE
AutoCompleteBox and ComboBox

### DISCUSSION THREAD

**Discussion by jim-jiang-github** [AvaloniaUI/Avalonia#11533] [Q&A] [answered]
How to implement AutoCompleteBox with a small button on right, which will expand the list after clicking. A fusion similar to combobox and AutoCompleteBox？

**Comment by sccombp** [comment, marked as answer]
It is possible but the AutoCompleteBox is a bit of a pain to work with right now.
I have attached a sample project that adds a button (as well as enabling the dropdown when no text has been entered) [TestAutoCompleteBehaviour_v11.zip](https://github.com/AvaloniaUI/Avalonia/files/11620393/TestAutoCompleteBehaviour_v11.zip)

**Reply by jim-jiang-github** [reply]
Thank you very much for your idea, this solved my problem well.

**Reply by buzzware** [reply]
I used the example provided here to create this test project combining outline material styling for AutoCompleteBox and the ability to drop the list by clicking a button
https://github.com/buzzware/AutoCompleteBoxPlay
Also see 
https://github.com/AvaloniaCommunity/Material.Avalonia/issues/370

Note that it works with ItemSource directly and using Populating+PopulateComplete, but not using AsyncPopulator (the drop list flashes once and disappears) 

I created this issue https://github.com/AvaloniaUI/Avalonia/issues/15416 - you might like to vote for it @jim-jiang-github jim-jiang-github
